### ISSUE TITLE
Autostep meta-update NaN-poisons normalizers and step-sizes past the v>0 gate (same class as #42)

### ISSUE THREAD

**Issue by ss251** [SlopDotCash/asi#55]
## Contribution provenance

- AI assistance: yes
- AI provider/model: xai / grok-4.6 (sibling-guard audit, reproductions, and fix in a local lane); anthropic / claude-fable-5 (independent verification and publish)
- Client / agent tooling: Grok Build CLI; claude-code
- Attribution status: self-reported

**Describe the bug**

Autostep has the same NaN-poisoning class fixed for IDBD in #42/#43 — its guards are not finiteness guards. Two leaks, both permanent:

1. **Zero-trace / overflow → NaN normalizers.** The Table-1 meta-gradient is the same left-associative `delta * x * h`; with fresh traces (`h=0`), an inf error (or a float32 overflow of the finite product) makes it NaN. Eq. 4 then writes `v = max(NaN, v_update) = NaN` (`jnp.maximum` propagates NaN), and `max(inf, NaN)` on every later finite update keeps it NaN — the normalizers never recover. (Grimly: `NaN > 0` is False, so the v>0 gate *accidentally* protects alpha on that first step while v dies.)
2. **Warm traces + inf → NaN alpha through the safety clip.** After a few finite updates `v` is finite nonzero; an inf error makes the meta-gradient inf, `v = max(inf, …) = inf`, the gate passes (`inf > 0`), and Eq. 5 computes `alpha * exp(mu * inf / inf) = alpha * exp(NaN) = NaN`. The "numerical safety" `clip(NaN, 1e-8, 1.0)` is NaN. Bias twin identical; `step_sizes`/`bias_step_size` stay NaN for every later finite update.

Not a hole (checked): zero-normalizer division — `v=0` is excluded by the gate and the `maximum(v, 1e-38)` floor is never consulted on that branch. IDBD under identical inputs (post-#43) stays finite and recovers. `AutostepGTDLambda.update` delegates to this path and inherits the fix.

**To Reproduce** (at `origin/main` `da8b86cd`)

```
uv run python -c '
import jax.numpy as jnp
from alberta_framework import Autostep
opt = Autostep(initial_step_size=0.01, meta_step_size=0.01)
x = jnp.ones(2, dtype=jnp.float32); st = opt.init(2)
r = opt.update(st, jnp.array(jnp.inf, dtype=jnp.float32), x)
print(r.new_state.normalizers)        # [nan nan]
st_w = st
for _ in range(5):
    st_w = opt.update(st_w, jnp.array(1.0, dtype=jnp.float32), x).new_state
r_w = opt.update(st_w, jnp.array(jnp.inf, dtype=jnp.float32), x)
print(r_w.new_state.step_sizes)       # [nan nan] — through the safety clip
'
```

Controls: finite errors on the same states leave everything finite; the overflow case poisons only the overflowed channel.

**Expected behavior**

A non-finite meta-gradient skips adaptation for that weight (previous normalizer and step-size kept — the #43 contract); the Eq. 5 "safety" clip must leave step-sizes inside `[1e-8, 1.0]` (finite); later finite updates must not inherit NaN state.

**Additional context**

Fix PR follows immediately: guard Eq. 4 with `where(isfinite(meta_gradient), …, v_prev)` and AND the Eq. 5 gate with the same finiteness check, all three sites — guard-only, with finite trajectories verified bit-identical to main (sha256 probe over step_sizes + normalizers + traces), per the #43 review precedent.
